homeassistant/components/flexmeasures/services.py

```diff
@@ -80,34 +83,40 @@ async def trigger_and_get_schedule(
         tzinfo = dt_util.get_time_zone(hass.config.time_zone)
         start = time_ceil(datetime.now(tz=tzinfo), resolution)
 
+        input_arguments = {
+            "sensor_id": get_from_option_or_config("power_sensor", entry),
+            "start": start,
+            "duration": get_from_option_or_config("schedule_duration", entry),
+            "soc_unit": "MWh",
+            "soc_min": get_from_option_or_config("soc_min", entry),
+            "soc_max": get_from_option_or_config("soc_max", entry),
+            "consumption_price_sensor": get_from_option_or_config(
+                "consumption_price_sensor", entry
+            ),
+            "production_price_sensor": get_from_option_or_config(
+                "production_price_sensor", entry
+            ),
+        }
+
+        LOGGER.info(input_arguments)
+
         schedule = await client.trigger_and_get_schedule(
-            sensor_id=config_data["power_sensor"],
-            start=start,
-            duration=config_data["schedule_duration"],
-            soc_unit=config_data["soc_unit"],
-            soc_min=config_data["soc_min"],
-            soc_max=config_data["soc_max"],
-            consumption_price_sensor=config_data["consumption_price_sensor"],
-            production_price_sensor=config_data["production_price_sensor"],
-            soc_at_start=call.data.get("soc_at_start"),
+            soc_at_start=call.data.get("soc_at_start"), **input_arguments
         )
 
-        schedule_state = start.isoformat()
         schedule = [
             {"start": start + resolution * i, "value": value}
             for i, value in enumerate(schedule["values"])
         ]
 
-        hass.states.async_set(
-            f"{DOMAIN}.charge_schedule",
-            new_state=schedule_state,
-            attributes={"schedule": schedule},
-        )
+        hass.data[DOMAIN]["schedule"]["schedule"] = schedule
+        hass.data[DOMAIN]["schedule"]["start"] = start
```

Comment:
We might want to store the schedule's duration, too, in order to store a fully self-contained description of the schedule. Otherwise, the interpretation of the schedule relies on the sensor resolution.